### ISSUE TITLE
Crest::BadRequest using Coveralls > v0.3.5

### ISSUE THREAD

**Issue by pvheiningen** [coverallsapp/coverage-reporter#80]
### What was wrong?
Hi there, we're using the universal coverage reporter to upload gcov coverage files to coveralls.io. We have some issues with newer versions. Any version above 0.3.5 gives me the `<Crest::BadRequest:HTTP status code 400: Bad Request>` error.

v0.3.5:
```
🚀 Posting coverage data to https://coveralls.io/api/v1/jobs
---
✅ API Response: {"message":"Job ##11.1","url":"https://coveralls.io/jobs/122307634"}
- 💛, Coveralls
```

v0.3.6:
```
🚀 Posting coverage data to https://coveralls.io/api/v1/jobs
#<Crest::BadRequest:HTTP status code 400: Bad Request>
Unhandled exception: HTTP status code 400: Bad Request (Crest::BadRequest)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
```

I believe the problem is potentially caused by the request being too large, if I limit the number of gcov files that we're uploading it starts working again on v0.3.6.

**Comment by mrexox** [comment]
Hey! Thank you for the issue. I have a few questions to better understand the issue:
- Did you try latest v0.4.3 version?
- How many files you are sending?

It looks like the problem is on a backend side, maybe it's too big payload. Could you share your gcov coverage file, please? You can send it on my email - valentin@coveralls.io, it would be the best way for me to find the issue.

**Comment by pvheiningen** [comment]
- Our CI system automatically pulled the latest version v0.4.3, which is how we found out that it didn't work anymore. It looks like it got broken in v0.3.6.
- We are sending 351 gcov files. 

Sent you an email with the actual gcov files, hopefully you can use those to reproduce.

**Comment by mrexox** [comment]
Hey! Thank you very much for sharing the reports. I found an issue and I'm releasing a fix in v0.4.4 of coverage reporter. If you use Github action or download the latest binary, nothing should be done on your side, just re-run the workflow. Otherwise, specify v0.4.4 version for the new workflows, please.

**Comment by pvheiningen** [comment]
Thanks a lot for the quick turnaround! I appreciate it! :tada:
